File: src/main/java/edu/projetJava/services/TestGemini.java
package edu.projetJava.services;

import edu.esportify.config.EnvConfig;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class TestGemini {
    private static final String DEFAULT_API_KEY = "AIzaSyBny7McZPfvUKJqlTNj2UW5H_kF5yaQB-w";
    public static void main(String[] args) {
        String m = "gemini-flash-latest";
        System.out.println("Testing " + m);
        try {
            String apiKey = EnvConfig.get("TEST_GEMINI_API_KEY", EnvConfig.get("GEMINI_API_KEY", DEFAULT_API_KEY));
            URL url = new URL("https://generativelanguage.googleapis.com/v1beta/models/" + m + ":generateContent?key=" + apiKey);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("POST");
            conn.setRequestProperty("Content-Type", "application/json");
            conn.setDoOutput(true);
            String body = "{\"contents\":[{\"parts\":[{\"text\":\"Hello\"}]}]}";
            conn.getOutputStream().write(body.getBytes());
            int rc = conn.getResponseCode();
            System.out.println("Code: " + rc);
            if (rc != 200) {
                BufferedReader in = new BufferedReader(new InputStreamReader(conn.getErrorStream()));
                String line;
                while ((line = in.readLine()) != null) {
                    System.out.println(line);
                }
            } else {
                BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
                String line;
                while ((line = in.readLine()) != null) {
                    System.out.println(line);
                }
            }
        } catch (Exception e) { e.printStackTrace(); }
    }
}

File: test_gemini.py
import urllib.request
import json

API_KEY = "AIzaSyBny7McZPfvUKJqlTNj2UW5H_kF5yaQB-w"
url = f"https://generativelanguage.googleapis.com/v1beta/models/gemini-flash-latest:generateContent?key={API_KEY}"
data = {
    "contents": [{"parts": [{"text": "Hello"}]}]
}

req = urllib.request.Request(url, json.dumps(data).encode("utf-8"), headers={"Content-Type": "application/json"})
try:
    with urllib.request.urlopen(req) as f:
        res = f.read()
        print(res.decode("utf-8"))
except urllib.error.HTTPError as e:
    print(e.code)
    print(e.read().decode("utf-8"))
